sisyphus/global_settings.py

```diff
@@ -317,69 +318,46 @@ def file_caching(path):
 #: Raise an exception if a Variable is accessed which is not set yet
 RAISE_VARIABLE_NOT_SET_EXCEPTION = False
 
+# Parameter used for debugging or profiling
+MEMORY_PROFILE_LOG = None
+
+USE_UI = True
+
+# Set to False to disable Warning of unset engine
+ENGINE_NOT_SETUP_WARNING = True
 
 # Internal functions
-def update_global_settings_from_text(text, filename):
-    """
-    :param str text:
-    :param str filename:
-    :return: nothing
-    """
-    # create basically empty globals
-    globals_ = {
-        '__builtins__': globals()['__builtins__'],
-        '__file__': filename,
-        '__name__': filename,
-        '__package__': None,
-        '__doc__': None,
-    }
-    globals_keys = list(globals_.keys())
-
-    # it might be useful to change the default environment by modifying the existing set/dict
-    # thus we need to add it to the globals, but not to globals_keys
-    globals_['DEFAULT_ENVIRONMENT_KEEP'] = DEFAULT_ENVIRONMENT_KEEP
-    globals_['DEFAULT_ENVIRONMENT_SET'] = DEFAULT_ENVIRONMENT_SET
-
-    # compile is needed for a nice trace back
-    exec(compile(text, filename, "exec"), globals_)
-
-    for k, v in globals_.items():
-        if k not in globals_keys:
-            globals()[k] = v
+ENVIRONMENT_SETTINGS = {}
+ENVIRONMENT_SETTINGS_PREFIX = 'SIS_'
 
-    if AUTO_SET_JOB_INIT_ATTRIBUTES:
-        import logging
-        logging.warning('AUTO_SET_JOB_INIT_ATTRIBUTES is deprecated, please set the attributes manually '
-                        'you might want to use self.set_attrs(locals())')
+GLOBAL_SETTINGS_FILE_CONTENT = ''
```

Comment:
Is this purely for the test case? Or would this have any other use?

If purely for the test case, I would maybe make it private (add `_` prefix, and not uppercase).

Also, in any case, add a small comment.

The logic below is also slightly wrong for it when there are multiple setting files. Only the last content would be in there.

Reply:
> The logic below is also slightly wrong for it when there are multiple setting files. Only the last content would be in there.

As `GLOBAL_SETTINGS_FILE_CONTENT = f"{GLOBAL_SETTINGS_FILE_CONTENT}` + "new settings file content", all files are present.

otherwise agree to your comment,

Reply:
Ah ok I missed that.

Reply:
It allows to log the exact settings which where used. I use to store them when creating my final report. I'm not doing it right now, but I still think it could be useful.

I changed the code to make it a little clearer that the new content is added to the original string.

Reply:
Ok for me then.